src/plugin/adapter-registry/c-tracker.ts

```diff
@@ -77,4 +78,8 @@ export class CTracker extends AdapterVariableTracker {
         const response = await sendRequest(session, 'evaluate', { expression: CEvaluateExpression.sizeOf(variableName), context: 'watch', frameId: this.currentFrame });
         return notADigit.test(response.result) ? undefined : BigInt(response.result);
     }
+
+    protected isPointer(variable: DebugProtocol.Variable): boolean {
+        return (!variable.type || variable.type.endsWith('*')) && (`0x${Number(variable.value).toString(16)}` === variable.value);
```

Comment:
I think the `*` check will likely suffice, but if we want to include the check for numerical value, let's use one that accommodates at least bases 10 and 16, which is the informal convention we've arrived at for formats we generally want to support.

Reply:
I just tried this with one of our debug adapters. The check for a numerical value is a bit of a problem there.

Background is that we show the pointer type as part of the value in the Variables display. For example
![image](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/assets/24247651/98ad0da3-6105-47d4-a901-bc83f0f1206c)

`type` alone would however work, see the info in the variable decorator hover-over:
![image](https://github.com/eclipse-cdt-cloud/vscode-memory-inspector/assets/24247651/31d8637b-67d0-4d83-a4dc-1cc5986a7a54)

What is the motivation behind verifying that a pointer is displayed as a hex value?

Another question is why not providing a `type` would resolve to `isPointer(...) === true`?

Reply:
OK, I see now why it checks for a numerical value. It's passed to the address field of the new Memory Inspector instance as is.

On the other hand, `readMemory` request doesn't mandate that a `memoryReference` (and hence the address field value) must be a numerical value. It could as well be a symbol or an expression. In fact something we would like to utilize in future with out debug adapters to allow for example arithmetic expressions beyond the given offset field.

Reply:
Yes, in principle it could just be the variable itself. If we believe that `x` is a pointer to `y`, then `x` should be the same `&y`, which is what CDT GDB, at least, uses as the `memoryReference` for `y`.